Detectors/MUON/MCH/Simulation/src/Digitizer.cxx

```diff
@@ -60,62 +60,76 @@ void Digitizer::init()
 
 void Digitizer::process(const std::vector<Hit> hits, std::vector<Digit>& digits)
 {
+  digits.clear();
+  mDigits.clear();
+  mMCTruthContainer.clear();
+  
   //array of MCH hits for a given simulated event
+  //to be added: exception handling
   for (auto& hit : hits) {
-    processHit(hit, mEventTime);
+    int labelIndex = mMCTruthContainer.getIndexedSize();//index for this hit
+    int detID = hit.GetDetectorID();
+    if(detID<299){
+      processHit(hit, detID, mMuonresponse_stat1, mEventTime, labelIndex);
+    } else {
+      processHit(hit, detID, mMuonresponse_stat2, mEventTime, labelIndex);
+    }
+ 
+    MCCompLabel label(hit.GetTrackID(), mEventID, mSrcID);
+    //Todo:: check/update DPL workflow how EventID and srcID set
+    mMCTruthContainer.addElementRandomAccess(labelIndex,label);//ToDo: check if needed like this
+
+    auto labels = mMCTruthContainer.getLabels(labelIndex);
+    std::sort(labels.begin(), labels.end());
```

Comment:
this is a genuine question here : do you have to sort the labels for _each_ hit ?

Reply:
Hi,
I copied this from EMCal, however, i struggle to understand now, why it is done like that and not differently. In fact, if I understand correctly, neither in my implementation nor in the EMCal code the list labels are used in some way.
In any case, the matching between the digit and the MCtruth container is done with the index within the array/list. I will come back to it, I have to check it more carefully.

Reply:
Hi @aphecetche ,
I checked it. It is in the EMCal code, but I don't understand the purpose at all, and I think that the last two lines are not needed. 
The "addElementRandomAccess" however, I don't see how to avoid it since the index inside makes later the identification  (unless one introduces some bookkeeping in parallel in order to put inside at once in the end). 
For the moment, I would prefer to keep it like this, ask Sandro during the PR if this is the correct way of doing it and then, if it is the case, potentially modifying with something more performant later.